lib/Parse/ParseExpr.cpp

```diff
@@ -3492,24 +3493,21 @@ void Parser::validateCollectionElement(ParserResult<Expr> element) {
   if (!isa<ArrayExpr>(subscriptExpr->getBase()))
     return;
 
-  auto baseExpr = cast<ArrayExpr>(subscriptExpr->getBase());
-  auto index = subscriptExpr->getIndex()->getValueProvidingExpr();
+  auto arrayExpr = cast<ArrayExpr>(subscriptExpr->getBase());
 
-  if (!isa<IntegerLiteralExpr>(index))
-    return;
-  auto indexExpr = cast<IntegerLiteralExpr>(index);
-
-  int elementsCount = baseExpr->getNumElements();
-  int indexValue = atoi(indexExpr->getDigitsText().str().c_str());
+  auto startLocOfSubscript = subscriptExpr->getIndex()->getStartLoc();
+  auto endLocOfArray = arrayExpr->getEndLoc();
 
-  auto withinBounds = indexValue >= 0 && indexValue < elementsCount;
-
-  if (!withinBounds) {
-    auto diag = diagnose(subscriptExpr->getLoc(),
-                         diag::subscript_collection_element_invalid_index,
-                         indexExpr->getDigitsText());
+  if (L->getLocForEndOfToken(SourceMgr, endLocOfArray) != startLocOfSubscript) {
+    auto diag =
+        diagnose(subscriptExpr->getLoc(), diag::subscript_array_element);
     diag.highlight(subscriptExpr->getSourceRange());
-    element.setIsParseError();
+    diag.fixItInsertAfter(endLocOfArray, ",");
```

Comment:
Fix-its can *either* be on the warning, *or* on multiple notes. A fix-it directly on a diagnostic indicates that that's only reasonable choice, which isn't the case here.

Reply:
Yeah I fixed that (this is an old commit)